lua/entities/gmod_wire_light.lua

```diff
@@ -244,20 +250,28 @@ function ENT:Setup(directional, radiant, glow, brightness, size, r, g, b, sprite
 		self.spritesize = math.Clamp( self.spritesize, 0, 256 )
 	end
 
+	if not self.on then self:SetOn(true) end
+	
 	self:Directional( self.directional )
 	self:Radiant( self.radiant )
 	self:SetGlow( self.glow )
 	self:SetBrightness( self.brightness )
 	self:SetSize( self.size )
 	self:SetSpriteSize( self.spritesize )
 
+	local inputs = {"Red", "Green", "Blue", "RGB [VECTOR]", "GlowBrightness", "GlowSize", "SpriteSize"}
 	if self.glow then
-		WireLib.AdjustInputs(self, {"Red", "Green", "Blue", "RGB [VECTOR]", "GlowBrightness", "GlowSize", "SpriteSize"})
-	else
-		WireLib.AdjustInputs(self, {"Red", "Green", "Blue", "RGB [VECTOR]", "SpriteSize"})
+		table.insert(inputs, 5, "GlowBrightness")
+		table.insert(inputs, 6, "GlowSize")
 	end
+	
+	if self.on then
+		table.insert(inputs, 1, "On")
+	end
```

Comment:
This doesn't make sense, there should always be an 'On' input.

Reply:
My intention was that the light having an "On" input is optional, since lights did not include that feature originally.

Reply:
I may have made this needlessly confusing but if `self.on` is false then `self:GetOn()` is true, which means lights are always lit. Otherwise, if `self.on` is true, then lights are conditionally lit based on the Wire "On" input. See [line 253](https://github.com/wiremod/wire/pull/2596/files/8c294dcd08706cd380158d37412ed02da04f87be#diff-e7da65d22a22a881d4154fffdff2ec5c461707c9a5645882ca7ca19ca25ff64fR253).